### PR TITLE
Add input validation

### PR DESCRIPTION
**Pull request by ao-armen**

### CODE REVIEW

**Comment by hpilosyan** [comment]
Why do you have multiple commits?

File: src/containers/App/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import logo from '../../styles/svgs/logo.svg';
+import appSchema from '../../validations/app';
+import useForm from 'react-hook-form';
```
Comment (hpilosyan) [inline src/containers/App/index.js]: What's the advantage of this over formik?

Comment (ao-armen) [reply src/containers/App/index.js]: It works faster, it's easier to read, it's based on React hooks, prevents unnecessary re-renders

Comment (hpilosyan) [reply src/containers/App/index.js]: As far as I know formik uses hooks under the hood as well

Comment (hpilosyan) [reply src/containers/App/index.js]: Also the question is if it gives us all the features we want

Comment (ao-armen) [reply src/containers/App/index.js]: But Formik re-renders much more then React hook form.
Yes, we have all the features we'll need.

Comment (ao-tigran) [reply src/containers/App/index.js]: Let's have a good discussion around this, @ao-armen has found some interesting stuff. Armen was looking at this, seems pretty intereseting

https://react-hook-form.com/

File: src/containers/App/index.js [hunk 2]
```diff
@@ -1,21 +1,48 @@
...
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Learn React
+        </a>
+
+        <br />
+
+        <input name="firstName" type="text" ref={register}></input>
+        {errors.firstName && 'First name is required.'}
```
Comment (hpilosyan) [inline src/containers/App/index.js]: We probably need a Validation component

**Pull request closed without merging**